src/cursor.rs

```diff
@@ -0,0 +1,171 @@
+use core::fmt;
+use std::{error::Error, sync::Arc};
+
+use crate::{icon::PIXEL_SIZE, platform_impl::PlatformCustomCursor};
+
+#[derive(Debug, Clone)]
```

Comment:
Is `Clone` actually necessary? We could just take a reference in `Window::set_custom_cursor()`.

Reply:
It says `error[E0521]: borrowed data escapes outside of method` so I think arc or something similar might be required.

Reply:
Try using `maybe_wait_on_main()` instead.

Reply:
On second thought, it might be better to keep the `Arc` but still take a reference. So just take a reference from the user but then clone the underlying `Arc` inside `Window::set_custom_cursor()`, we can decide to remove the `Arc` later if we still think that's better.

Reply:
Sound good